### PR TITLE
ICD 9 V Code Prediction Task and Example

### PR DESCRIPTION
**Pull request by sharimk2**

### CODE REVIEW

**Review by jhnwu3 — CHANGES_REQUESTED** [review]
Hey Sharim! Thanks for the help! Can we implement some unit tests sorta like how paul does it here?

https://github.com/sunlabuiuc/PyHealth/blob/master/tests/core/test_sdoh.py

Basically, it helps to just test that the class itself works. 

Other things would be adding to the docs/ (Cursor does a pretty good job here given some examples).

File: pyhealth/models/sdoh_icd9_task.txt [hunk 1]
```diff
@@ -0,0 +1,158 @@
+# SDOH Medical Coding Task
```
Comment (jhnwu3) [inline pyhealth/models/sdoh_icd9_task.txt]: Anyway to refactor this as a *.py file or just have the prompt here in the llm.py file? 

I guess another thing we could do is have a sdoh_icd9_llm/ directory for your work here.

File: pyhealth/models/sdoh_icd9_llm.py [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+PROMPT_PATH = os.path.join(os.path.dirname(__file__), "sdoh_icd9_task.txt")
+
+
+def _load_prompt_template() -> str:
+    with open(PROMPT_PATH, "r", encoding="utf-8") as f:
+        return f.read()
+
+
+class SDOHICD9LLM:
+    """Admission-level SDOH ICD-9 V-code detector using an LLM."""
```
Comment (jhnwu3) [inline pyhealth/models/sdoh_icd9_llm.py]: Can we add more doc strings on how a user might use this model here?

Comment (jhnwu3) [reply pyhealth/models/sdoh_icd9_llm.py]: Please see some examples: https://github.com/sunlabuiuc/PyHealth/blob/master/pyhealth/models/stagenet.py

**Review by jhnwu3 — CHANGES_REQUESTED** [review]

File: pyhealth/models/sdoh_icd9_llm.py (hunk 2 above)
Comment (sharimk2) [reply pyhealth/models/sdoh_icd9_llm.py]: I added more doc strings, let me know what you think @jhnwu3

File: pyhealth/datasets/sdoh_icd9.py [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+}
+
+
+class SDOHICD9NotesDataset:
+    """CSV-backed dataset for SDOH ICD-9 V-code detection from notes."""
+
+    def __init__(
+        self,
+        csv_path: str,
+        dataset_name: Optional[str] = None,
```
Comment (jhnwu3) [inline pyhealth/datasets/sdoh_icd9.py:30]: Is this your own dataset or did you use MIMIC-III? Or do you have to download it from somewhere?

File: pyhealth/datasets/mimic3.py [hunk 4]
```diff
@@ -85,3 +89,156 @@ def preprocess_noteevents(self, df: pl.LazyFrame) -> pl.LazyFrame:
             .alias("charttime")
         )
         return df
+
+
+class MIMIC3NoteDataset:
```
Comment (jhnwu3) [inline pyhealth/datasets/mimic3.py:94]: I believe MIMIC3Dataset already supports the loading of notes iirc, so this feels a bit redundant.

File: pyhealth/tasks/sdoh_icd9_detection.py [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+
+import torch
+
+from .base_task import BaseTask
+from .sdoh_utils import TARGET_CODES, codes_to_multihot
+
+
+class SDOHICD9AdmissionTask(BaseTask):
+    """Builds admission-level samples for SDOH ICD-9 V-code detection.
+
```
Comment (jhnwu3) [inline pyhealth/tasks/sdoh_icd9_detection.py:11]: Can we get docstrings on how to use this task. For reference, the .set_task() feature here that you shouldn't overwrite unless you really need to in each of the BaseDataset classes require you pass a task through a dataset.set_task(TaskClass) here as it will get parallelized and used by our backend here.

**Review by jhnwu3 — CHANGES_REQUESTED** [review]
Hey Sharim,

I would recommend checking out the pyhealth.datasets and pyhealth.tasks tutorials here: https://pyhealth.readthedocs.io/en/latest/tutorials.html to better understand how to extend PyHealth with your own dataset.

I think if you have separate labels in your dataset that's stored somewhere else on top of MIMIC3, then we'd probably need to create a new Dataset(BaseDataset) class here, but otherwise if its just a MIMIC3 showcase, I would recommend using the existing APIs here.
